pubpub/platform
# refactor: improve readability and typesafety in permissionField

- **refactor: rewrite the API token permission field logic to be correct and more readable**

## Issue(s) Resolved

## High-level Explanation of PR

I was trying to fix a weird CI error in #646 , which lead me to restructure a bunch of this stuff in `PermissionField.tsx` to actually work and be slightly easier to understand.

## Test Plan
1. Create an API token
2. Make a request
3. Should work as normal.

## Screenshots (if applicable)

## Notes
